robot_smach_states/src/robot_smach_states/human_interaction/recognize_person.py

```diff
@@ -0,0 +1,79 @@
+# System
+import typing
+
+# ROS
+import PyKDL as kdl
+import rospy
+import smach
+
+# TU/e Robotics
+from image_recognition_msgs.msg import CategoricalDistribution, CategoryProbability, Recognition
+from robot_skills.robot import Robot
+
+# Robot Smach States
+from robot_smach_states.util.designators.core import VariableDesignator
+
+
+def get_operator_name(recognition: Recognition, threshold: typing.Optional[float] = 3.0) -> str:
+    """
+    Gets the operator name from a recognition
+
+    :param recognition: recognition of the face
+    :param threshold: if the best match is below this threshold, it is designated as unknown and an exception is raised
+    :return: operator name
+    :raises: RuntimeError
+    """
+    distribution = recognition.categorical_distribution  # type: CategoricalDistribution
+    if not distribution.probabilities:
+        raise RuntimeError("Recognition does not contain probabilities")
+    best_match = max(distribution.probabilities, key=lambda cp: cp.probability)  # type: CategoryProbability
+    if best_match.probability < threshold:
+        raise RuntimeError(
+            f"Best match has a probability of {best_match.probability} which is lower than the threshold {threshold}"
+        )
+    return best_match.label
+
+
+class RecognizePerson(smach.State):
```

Comment:
can we make the distinction between getting no detections i.e. not seeing anyone and seeing someone but not recognising them? Would that be something desirable?

Reply:
Perhaps, I added the distinction: 2f723d8ac56b36b342aabb56cb9459e0ff16ebc0